### ISSUE TITLE
"Save and new" action for form fields only works for admin users

### ISSUE THREAD

**Issue by plorenz-etes** [contao/contao#414]
**Affected version(s)**

4.4.36, but probably all versions until now

**Description**  

Creating a form field as a non-administrative user using the "Save and new" feature produces a "Forbidden" error

**How to reproduce**  

-Log in using a non-admin account
-Open the form generator
-Go to the list of form fields of an existing form or create a new one and edit its fields
-Add a new form field of any type
-Instead of clicking "Save" or "Save and close", click on the tiny arrow on the right side of the "Save and close" button and select "Save and create"

You should now see the following error page:

```
Forbidden
What's the problem?

Not enough permissions to access form ID .
```

**Additional information**

As you can see, the error message is missing the ID of the form. The error is thrown in the core-bundle's `dca/tl_form_field.php` in function `checkPermissions()`.

Administrators don't run into trouble here because permissions won't be checked at all. For non-admins, the following code snippet causes the error:

```
			case 'create':
				if (!\strlen(Contao\Input::get('id')) || !\in_array(Contao\Input::get('id'), $root))
				{
					throw new Contao\CoreBundle\Exception\AccessDeniedException('Not enough permissions to access form ID ' . Contao\Input::get('id') . '.');
				}
				break;
```

`Contao\Input::get('id')` returns an empty string because there is no ID set in the URL, which for example looks like this:

<domain>/contao?do=form&table=tl_form_field&act=create&mode=1&pid=19&rt=<token>

**Comment by leofeyer** [comment]
Which form permissions did you configure for the user or user group?

**Comment by plorenz-etes** [comment]
The following form rights have been configured:

- In `Back end modules`, `Content` / `Form generator` is selected
- In `Form permissions`, all available forms are selected under `Allowed forms`
- In `Form permissions`, both `Create` and `Delete` permissions have been granted
- In `Allowed fields`, all fields of the tables `tl_form` and `tl_form_field` have been selected

**Comment by leofeyer** [comment]
Fixed in b3fb6806a25d1672cbb908825bf4b04caef2d9cf.

**Comment by plorenz-etes** [comment]
Thanks! :)
